desktop-app/src/renderer/components/DeviceManager/DeviceDetailsModal.tsx

```diff
@@ -48,12 +48,14 @@ const DeviceDetailsModal = ({
       setIsTouchCapable(device.isTouchCapable);
       setIsMobileCapable(device.isMobileCapable);
     } else {
-      setName('');
-      setWidth(400);
-      setHeight(600);
-      setUserAgent('');
-      setType('phone');
-      setDpr(1);
+      setTimeout(() => {
+        setName('');
+        setWidth(400);
+        setHeight(600);
+        setUserAgent('');
+        setType('phone');
+        setDpr(1);
+      }, 500);
```

Comment:
using setTimeout is not a good practice in general, I think that this might be more of a race condition issue. Have you looked at the order of the setStates?